### PR TITLE
feat(graph): add profile view

### PR DESCRIPTION
**Pull request by gaspardthrl**
# What I did
I implemented the user interaction flow in the graph view that is:

- The screen focuses on the clicked node by _centering_ and _zooming_  through an **animation**
- When the animation is _triggered_, a **modal** fades in and let us see the user's profile picture and its name (Note that other functionalities will arrive after the database integration - e.g, Button to navigate to the user's profile page, Main interests, ...)
- Clicking outside the modals dismisses it.
- I added different profile pictures for better visualization of different rendering especially when displaying the user's profile picture on the modal
- I implemented tests and achieved up to 98.6% code coverage
- Merged main to avoid conflicts

# How I did it

- For the node _centering_ and _zooming_ **animation** I built a function called `nodeZoomIn` that takes the _clicked node_ of type `Node` as arguments. It then triggers the necessary hooks to disable gestures during animation and sets the modal to visible. It then interpolates over a predefined number of **frames** using `requestAnimationFrame` which takes a callback function as argument. After the last frame, it triggers the necessary hooks to enable gestures.
- Regarding the Modal implementation, I used the `Modal` component from React-Native
- I used the `TouchableWithoutFeedback` component to dismiss the modal when clicking outside of it
- For the tests, I mainly built `async` functions with `await`, `act` and also made use of jest's Timers `jest.useFakeTimers()`, `jest.useRealTimers()`, `jest.advanceTimersByTime(...)`. Finally I made use of `fireEvent` to test when a user presses a component (both in and out).
 
# How to verify it

1. First replace the App.tsx file with the following code:
```javascript
import { NavigationContainer } from "@react-navigation/native"
import React, { useEffect } from "react"
import * as SplashScreen from "expo-splash-screen"
import RegistrationStackNavigator from "./navigation/Registration/RegistrationStackNavigator"
import { GestureHandlerRootView } from "react-native-gesture-handler"
import styles from "./styles"
import {
  useFonts,
  JetBrainsMono_100Thin_Italic,
  JetBrainsMono_400Regular,
  JetBrainsMono_700Bold,
} from "@expo-google-fonts/jetbrains-mono"
import { SafeAreaProvider } from "react-native-safe-area-context"
import { StatusBar } from 'expo-status-bar'
import ExampleForceDirectedGraph from "./components/Graph/ForceDirectedGraph/ExampleForceDirectedGraph"

SplashScreen.preventAutoHideAsync()

export default function App() {
  const [fontsLoaded] = useFonts({
    JetBrainsMono_100Thin_Italic,
    JetBrainsMono_400Regular,
    JetBrainsMono_700Bold,
  })

  useEffect(() => {
    async function hideSplashScreen() {
      if (fontsLoaded) {
        await SplashScreen.hideAsync()
      }
    }
    hideSplashScreen()
  }, [fontsLoaded])

  if (!fontsLoaded) {
    return null
  }
  return (
    <GestureHandlerRootView style={styles.gestureHandler}>
      <ExampleForceDirectedGraph />
    </GestureHandlerRootView>
  )
}
```
2. Try dragging the graph without clicking on any nodes to ensure that we can still move the whole graph.
3. Press a node and drag it, without releasing the press, to verify that nodes can still be dragged individually.
4. Perform a short press on a node (should be less than 100ms - so just tap the node basically) and check that the graph moves by itself and zooms on the node while witnessing the apparition of a modal with the correct profile picture and name (If you are already at zoom level 2 and / or the node you click is already at the center of the screen, you will only see the modal appear but won't notice the animation. To be sure to witness the animation dezoom the graph and click on an uncentered node).
5. When the modal is displayed try clicking outside of it, the modal should disappear and you should be able to interact with the graph again.
6. To run the tests, run `npm run test` in your terminal.

# Demo video
https://github.com/uniconnect-epfl/uniconnect/assets/81919237/74722411-7a22-4e3c-affb-aab0101ea6e0

# Pre-merge checklist
The changes I have introduced:
- [x] work correctly
- [x] do not break other functionalities
- [x] work correctly on Android
- [x] are fully tested

### CODE REVIEW

**Comment by GustaveCharles** [comment]
Amazing job, the graph and animations are so so smooth! And you even managed to test everything. Two small remarks: 
- when you say "I added different profile pictures", you refer to the different colors of the node right?
- the UI is almost perfect, there's one thing that I have noticed, is that sometimes, when you want to drag a node, it accidentally interprets it as if I clicked on the node. This is a very minor problem and should maybe not be addressed.

**Comment by gaspardthrl** [comment]
> Amazing job, the graph and animations are so so smooth! And you even managed to test everything. Two small remarks:
> 
> * when you say "I added different profile pictures", you refer to the different colors of the node right?
> * the UI is almost perfect, there's one thing that I have noticed, is that sometimes, when you want to drag a node, it accidentally interprets it as if I clicked on the node. This is a very minor problem and should maybe not be addressed.

Thanks!

Yes regarding the drag / click problem I decided for now to differentiate the two as follows:
- If you press the node for more than N milliseconds, it is interpreted as a "drag" action.
- However, if you press the node for less than M milliseconds (e.g., you drag the node very very fast like when you swipe), it is interpreted as a "drag" action also.
- Otherwise it is interpreted as a click.

I am aware that it might not be the best solution and I think that it will be replaced by some other logic like the total displacement resulting from the touch (i.e., extremely small displacement means you did not move the node and therefore it is a click). I might event combine the two to say that if displacement is very small but press lasts more than X milliseconds nothing happens.

Let me know what you think!

**Review by AlbertoCentonze — CHANGES_REQUESTED** [review]
Good job, very impressed by the tests and the overall quality of the code.

Regarding the comments to ignore ESlint on `require`, I still think it's a good idea not to disable this rule globally, as hopefully, you're the only one that needs to use the old `require` syntax.
However, I think we could go for a solution that does not pollute the code as much as line-by-line comments do.
Here two possible solutions:
1. Ignore specific files in `.eslintrc.js`, here's an example:
```javascript
"overrides": [
    {
      "files": ["path/to/file.js"],
      "rules": {
        "no-console": "off"
      }
    }
  ]
```
2. Ignore whole folders in `.eslintrc.js`. Here's an example:
```javascript
"overrides": [
  {
    "files": ["commonComponents/**/*.js"],
    "rules": {
      "import/prefer-default-export": "off"
    }
  }
]
```

Also, I noticed that you still haven't checked the "works on android" in the pre-merge checklist. Do you need any help with that?

**Comment by gaspardthrl** [comment]
> ```js
> "overrides": [
>     {
>       "files": ["path/to/file.js"],
>       "rules": {
>         "no-console": "off"
>       }
>     }
>   ]
> ```

Okay I will modify the .eslintrc.js file directly.

Also regarding the "works on android" check I can't get the emulator to run so I will gladly accept your help!

**Comment by AlbertoCentonze** [comment]
https://github.com/uniconnect-epfl/uniconnect/assets/11707683/66966f59-fd08-4964-b049-0d2b6933a8e5

Seems to be working fine on the emulator! The animation is a bit laggy but I would safely assume this is not the case when running on real hardware. 

Maybe @GaelCondeLosada can help us out here. Could you run it on your phone?

**Review by GustaveCharles — APPROVED** [review]

File: __test__/components/Graph/Force-Directed-Graph/ForceDirectedGraph.test.tsx [hunk 1]
```diff
@@ -1,21 +1,291 @@
...
+  fireEvent,
+  waitFor,
+} from "@testing-library/react-native"
 import ForceDirectedGraph from "../../../../components/Graph/ForceDirectedGraph/ForceDirectedGraph"
 import Graph from "../../../../components/Graph/Graph"
+import { State } from "react-native-gesture-handler"
+
+jest.mock("react-native-gesture-handler", () => {
+  // eslint-disable-next-line
+  const View = require("react-native").View
```
[the remark this answers is not in the document]
Comment (gaspardthrl) [reply __test__/components/Graph/Force-Directed-Graph/ForceDirectedGraph.test.tsx:14]: You are right, I will use the better practice

**Comment by GustaveCharles** [comment]
> > Amazing job, the graph and animations are so so smooth! And you even managed to test everything. Two small remarks:
> > 
> > * when you say "I added different profile pictures", you refer to the different colors of the node right?
> > * the UI is almost perfect, there's one thing that I have noticed, is that sometimes, when you want to drag a node, it accidentally interprets it as if I clicked on the node. This is a very minor problem and should maybe not be addressed.
> 
> Thanks!
> 
> Yes regarding the drag / click problem I decided for now to differentiate the two as follows:
> 
> * If you press the node for more than N milliseconds, it is interpreted as a "drag" action.
> * However, if you press the node for less than M milliseconds (e.g., you drag the node very very fast like when you swipe), it is interpreted as a "drag" action also.
> * Otherwise it is interpreted as a click.
> 
> I am aware that it might not be the best solution and I think that it will be replaced by some other logic like the total displacement resulting from the touch (i.e., extremely small displacement means you did not move the node and therefore it is a click). I might event combine the two to say that if displacement is very small but press lasts more than X milliseconds nothing happens.
> 
> Let me know what you think!

I think that the system that you have actually works decently, and that there is no need to change it for now. This might be something for a V2. Once again, your work is amazing already!

**Pull request merged**